x/mongo/driver/batch_cursor.go

```diff
@@ -326,6 +329,40 @@ func calcGetMoreBatchSize(bc BatchCursor) (int32, bool) {
 	return gmBatchSize, true
 }
 
+// commentToBSONCoreValue will convert the input comment from an "any" type to
+// a bsoncore.Value. If the conversion cannot occur, this function returns nil.
+// If marshaling the comment for the bsoncore.Value Data fails, this comment
+// will return an error.
+func commentToBSONCoreValue(comment interface{}) (*bsoncore.Value, error) {
+	if comment == nil {
+		return nil, nil
+	}
+
+	// If the comment is already a bsoncore.Value, then do nothing.
+	if value, ok := comment.(bsoncore.Value); ok {
+		return &value, nil
+	}
+
+	refValue := reflect.ValueOf(comment)
+	if refValue.Kind() == reflect.Map && refValue.Len() > 1 {
+		return nil, nil
+	}
+
+	registry := bson.DefaultRegistry
+	buf := make([]byte, 0, 256)
+
+	bsonType, bsonValue, err := bson.MarshalValueAppendWithRegistry(registry, buf[:0], comment)
```

Comment:
I believe to retain parity with the existing `comment` behavior, we need to support marshaling with a user-provided BSON codec registry. A `Cursor` has a `registry` associated with it, but a `BatchCursor` currently does not have a registry passed to it.

Is it possible to move the `comment` marshal logic to the `Cursor` API? Otherwise, you may have to pass the `Cursor` registry to `BatchCursor`.

Reply:
Good point! 

The mongo.Cursor object has a user-defined registry. In order to transform the comment to a bsoncore.Value, we would need to either:

1. Set a cursor error in the comment setter. This is probably fine but AFAIK we do not check this error internally, so it would be on the user to do so. This seems unwise, as knowledge of this condition would require a comment, etc.
2. Return an error on the "SetComment" method. This is better than option 1 but it's kind of strange and an anti-pattern considering how setters work in the rest of the Go Driver.

Given how strange the above two options are, I'm going to opt to add a registry to the SetComment function signature on the batch cursor interface. LMK what you think.